tests/test_communication_utils.cpp

```diff
@@ -0,0 +1,196 @@
+/*
+  Copyright 2021 OPM-OP AS.
+
+  This file is part of the Open Porous Media project (OPM).
+
+  OPM is free software: you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation, either version 3 of the License, or
+  (at your option) any later version.
+
+  OPM is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with OPM.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+
+#include "config.h"
+
+#define BOOST_TEST_MODULE CommunicationUtilities
+#include <boost/test/unit_test.hpp>
+#include <boost/test/unit_test.hpp>
+#include <dune/common/parallel/mpihelper.hh>
+#include <opm/common/utility/CommunicationUtils.hpp>
+#include <vector>
+#include <random>
+
+class MPIError {
+public:
+  /** @brief Constructor. */
+  MPIError(std::string s, int e) : errorstring(s), errorcode(e){}
+  /** @brief The error string. */
+  std::string errorstring;
+  /** @brief The mpi error code. */
+  int errorcode;
+};
+
+#ifdef HAVE_MPI
+void MPI_err_handler(MPI_Comm *, int *err_code, ...){
+  char *err_string=new char[MPI_MAX_ERROR_STRING];
```

Comment:
Do you need to use the free store here?  What's a typical value of `MPI_MAX_ERROR_STRING`?  If it's just a couple of hundred bytes would you be able to use a `std::array<char,N>` here?  By the way, does `MPI_MAX_ERROR_STRING` include the terminating `\0` character?  If you **do** have to use the free store, could you use `unique_ptr<char[]>` instead?

Reply:
> Do you need to use the free store here?

Nope and I will change that (use a vector) just to make you happy.

> What's a typical value of `MPI_MAX_ERROR_STRING`? If it's just a couple of hundred bytes would you be able to use a `std::array<char,N>` here?

That seems implementation dependent and I do not want to look it up for the particular MPI implementation that I am using.

> By the way, does `MPI_MAX_ERROR_STRING` include the terminating `\0` character? If you **do** have to use the free store, could you use `unique_ptr<char[]>` instead?

As the [MPI method](https://www.mpich.org/static/docs/latest/www3/MPI_Error_string.html) does not tell anything about that, but return the length, I think one should assume that it is not terminated. Better safe than sorry.

I should have just skipped the error handler...

Reply:
Unsing std::vector now